sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TablesJacksonSerializer.java

```diff
@@ -27,71 +25,117 @@
         final JsonNode node = super.serializer().readTree(inputStream);
-        final Map<String, Object> rootObject = new HashMap<>();
+        Map<String, Object> singleValue = null;
+
         for (Iterator<Map.Entry<String, JsonNode>> it = node.fields(); it.hasNext();) {
             final Map.Entry<String, JsonNode> entry = it.next();
+            final String fieldName = entry.getKey();
             final JsonNode childNode = entry.getValue();
 
-            if (entry.getKey().equals(TableConstants.ODATA_METADATA_KEY)) {
+            if (fieldName.equals(TablesConstants.ODATA_METADATA_KEY)) {
                 odataMetadata = childNode.asText();
-                continue;
-            }
-
-            // Represents the entries in the response. It's possible that it is a single or multiple response.
-            if (entry.getKey().equals("value")) {
+            } else if (fieldName.equals("value")) {
                 if (childNode.isArray()) {
-                    logger.info("Node is an array of items.");
+                    // This is a multiple-entity response.
+                    for (JsonNode childEntry : childNode) {
+                        values.add(getEntityFieldsAsMap(childEntry));
+                    }
+                } else {
+                    // This is a single-entity response where the user just happened to use the key "value".
+                    if (singleValue == null) {
+                        singleValue = new HashMap<>();
+                    }
+                    singleValue.put(fieldName, getEntityFieldAsObject(node, fieldName));
                 }
-
-                throw logger.logExceptionAsError(
-                    new UnsupportedOperationException("Multiple return values not supported yet."));
+            } else {
+                // This is a single-entity response.
+                if (singleValue == null) {
+                    singleValue = new HashMap<>();
+                }
+                singleValue.put(fieldName, getEntityFieldAsObject(node, fieldName));
             }
-
-            rootObject.put(entry.getKey(), entry.getValue().asText());
         }
 
-        values.add(0, rootObject);
+        if (singleValue != null) {
+            if (values.size() > 0) {
+                throw logger.logExceptionAsError(new IllegalStateException(
+                    "Unexpected response format. Response containing a 'value' array must not contain other properties."
+                ));
+            }
+            values.add(singleValue);
+        }
 
         return (U) new TableEntityQueryResponse()
             .setOdataMetadata(odataMetadata)
             .setValue(values);
     }
 
-    @Override
-    public <U> U deserialize(HttpHeaders httpHeaders, Type type) throws IOException {
-        return super.deserialize(httpHeaders, type);
+    private Map<String, Object> getEntityFieldsAsMap(JsonNode node) {
```

Comment:
I think this can be made static. It doesn't rely on any state in the class, iirc.i

Reply:
Definitely doable, but I would have to have a static logger for those methods and I'd prefer to reuse the existing logger unless there's a significant benefit to using static methods here.